wapitiCore/attack/mod_csrf.py

```diff
@@ -0,0 +1,203 @@
+
+        # Replace anti-csrf token value from form with "wapiti"
+        mutated_post_params = [
+            param if param[0] != self.csrf_string else [self.csrf_string, "wapiti"]
+            for param in original_request.post_params
+        ]
+
+        # Replace anti-csrf token value from headers with "wapiti"
+        mutated_headers = original_request.headers
+        if self.csrf_string in mutated_headers:
+            mutated_headers[self.csrf_string] = "wapiti"
+
+        mutated_request = Request(
+            path=original_request.path, method=original_request.method,
+            get_params=original_request.get_params, post_params=mutated_post_params,
+            file_params=original_request.file_params, referer=original_request.referer,
+            link_depth=original_request.link_depth
+        )
+
+        original_response = self.crawler.send(original_request, follow_redirects=True)
+        mutated_response = original_response
+
+        try:
+            mutated_response = self.crawler.send(mutated_request, headers=mutated_headers, follow_redirects=True)
+
+        except ReadTimeout:
+
+            self.log_orange("---")
+            self.log_orange(Anomaly.MSG_TIMEOUT, original_request.path)
+            self.log_orange(Anomaly.MSG_EVIL_REQUEST)
+            self.log_orange(mutated_request.http_repr())
+            self.log_orange("---")
+
+            anom_msg = Anomaly.MSG_PARAM_TIMEOUT.format(self.csrf_string)
+
+            self.add_anom(
+                request_id=original_request.path_id,
+                category=Anomaly.RES_CONSUMPTION,
+                level=Anomaly.MEDIUM_LEVEL,
+                request=mutated_request,
+                info=anom_msg,
+            )
+
+        else:
+            if mutated_response.status == 500:
+                anom_msg = Anomaly.MSG_PARAM_500.format(self.csrf_string)
+
+                self.add_anom(
+                    request_id=original_request.path_id,
+                    category=Anomaly.ERROR_500,
+                    level=Anomaly.HIGH_LEVEL,
+                    request=mutated_request,
+                    info=anom_msg,
+                )
+
+                self.log_orange("---")
+                self.log_orange(Anomaly.MSG_500, original_request.path)
+                self.log_orange(Anomaly.MSG_EVIL_REQUEST)
+                self.log_orange(mutated_request.http_repr())
+                self.log_orange("---")
+
+            return not self.is_same_response(original_response, mutated_response)
+
+        return True
+
+
+    def attack(self):
+        forms = self.persister.get_forms(attack_module=self.name) if self.do_post else []
+
+        for original_request in forms:
+
+            if self.verbose >= 1:
+                print("[+] {}".format(original_request))
+
+            csrf_value = self.is_csrf_present(original_request)
+            # check if token is present
+            if not csrf_value:
```

Comment:
This is very noisy as it will print something for every possible value of the form (see example with Acunetix website below)

We should reduce to one message `per original_request.url` / `original_request.post_keys` tuple

```
---
Falsification de requête inter-site (CSRF)  en provanance de POST http://testphp.vulnweb.com/cart.php (4)
        data: price=986&addcart=3
[+] POST http://testphp.vulnweb.com/cart.php (4)
        data: price=500&addcart=1
---
Falsification de requête inter-site (CSRF)  en provanance de POST http://testphp.vulnweb.com/cart.php (4)
        data: price=500&addcart=1
[+] POST http://testphp.vulnweb.com/cart.php (4)
        data: price=460&addcart=5
---
Falsification de requête inter-site (CSRF)  en provanance de POST http://testphp.vulnweb.com/cart.php (4)
        data: price=460&addcart=5
[+] POST http://testphp.vulnweb.com/cart.php (4)
        data: price=15000&addcart=7
---
Falsification de requête inter-site (CSRF)  en provanance de POST http://testphp.vulnweb.com/cart.php (4)
        data: price=15000&addcart=7
[+] POST http://testphp.vulnweb.com/cart.php (4)
        data: price=1000&addcart=4
---
Falsification de requête inter-site (CSRF)  en provanance de POST http://testphp.vulnweb.com/cart.php (4)
        data: price=1000&addcart=4
[+] POST http://testphp.vulnweb.com/cart.php (4)
        data: price=800&addcart=2
---
Falsification de requête inter-site (CSRF)  en provanance de POST http://testphp.vulnweb.com/cart.php (4)
        data: price=800&addcart=2
[+] POST http://testphp.vulnweb.com/cart.php (4)
        data: price=10000&addcart=6
---
```